### DISCUSSION TITLE
trying to learn

### DISCUSSION THREAD

**Discussion by jayfranks** [community/community#13138] [Other Feature Feedback, Questions, & Ideas]
hey im trying to learn how to code and every time i google something its either the basics of basics that is boring af or im lead to this site. it may be obvious to people on here but why do i keep getting pulled here like its actually useful? does this site provide free courses or is it just a platform of botched crap that makes no sense because the layout is terrible and cant work out its head from its ass.

**Comment by dndus** [comment]
Listen, There you can get answer about your question, there are a lot of public projects, packets and etc. If you wanna learn, there are free courses [Udemy](https://www.udemy.com/) [Edx](https://www.edx.org) [Coursera](https://www.coursera.org) you can learn whatever you want. Good luck!

**Reply by jayfranks** [reply]
they are not helpful none of them explain anything they tell you what to write and then you complete something without any idea of what they just got you to do

**Reply by dndus** [reply]
Okay, but if don't watch lessons how you can learn, first complete these lessons then if you wanna built projects write here. I hope you find a team which is working with you. Practice, best way to learn.

**Reply by jayfranks** [reply]
i get what you mean but there is nothing out there that i can find that is educational its just do this do that and no explanation of anything. so by the end you spend 40 hours learning nothing because you dont understand any of what they had you do. ive waisted plenty of time doing all this suggested with 0 outcome because my brain needs to be able to understand the meanings and processes of everything its doing and all these do is tell you what to write

**Reply by jayfranks** [reply]
and so you know i looked at your suggested sites one wants my cash then will deliver a load of crap another is free but requires knowledge of the area and the other does not even cover kotlin

**Reply by delihomen-debug** [reply]
I think a lot of beginners run into this when learning to code. The problem is that search results often jump between extremely basic tutorials and advanced discussions without much middle ground. A good way to progress is usually to pick one small project, learn the concepts you need for that project, and build from there instead of trying to absorb random tutorials.

Documentation and community examples can feel confusing at first, but once you understand how to read them, they become much more useful. Interactive tools can also help because you can experiment and immediately see what your code is doing. For example, when learning older communication systems like Morse code, having a live converter to test ideas can make the concepts easier to understand: [https://morsecodetranslatr.com/](https://morsecodetranslatr.com/)

The biggest thing is finding resources that match your current level rather than just following whatever Google ranks first.

**Comment by zjeffer** [comment]
GitHub is a site to save & share code (to work together in a team for example), it has nothing to do with courses. You can view other people's projects and contribute.

If you want to learn how to code, I suggest YouTube, reading documentation, freecodecamp, etc...

**Reply by jayfranks** [reply]
youtube is not reliable been there tried that and them sites are a croc of ****

**Comment by ABSanthosh** [comment]
Hello there!

Let me just clear some stuff up for you:
1) This is a platform where people store their programming projects. This is based on a popular version controlling system called Git
2) The reason why you are being pulled to this site is because the "thing" that you are searching for is already done by someone here 
3) No, this isn't a platform that gives free courses but once you get used to the layout, you can learn a lot from how others write code.

**Reply by jayfranks** [reply]
so this shouldn't be a result on the internet not what i asked for

**Reply by ABSanthosh** [reply]
If that's how you feel, then I think its time for you to write your own search engine... A good place to start would be GitHub where you can search for people who decided to write their own search engine and draw inspirations.

Maybe the reason github shows up a lot in your searches is because you are searching for complete projects instead of concepts but hey, I don't know how you search

**Reply by jayfranks** [reply]
thats not a bad idea but i struggle understanding how any of this site works its just a giant mess for someone who isnt in this world. it might look like im being difficult im not it just isnt designed for people wanting to learn

**Comment by dmadisetti** [comment]
Just for the Github team, I think this is indication that your feedback button is ambiguous and/or misplaced.

but welcome @jayfranks!

**Reply by jayfranks** [reply]
well maybe you just should not be a result on search engines i google something and get this crap. like its anything to do with my search

**Reply by zjeffer** [reply]
10s of millions of people use GitHub, it being on Google searches is very expected. If you want to exclude a site from your Google searches, you should add `-inurl:github.com` to the end of your search.

You're being very hostile to people just because **you** don't understand this site. Seeing as you're trying to learn how to code, GitHub (or an alternative like GitLab) should be very useful to you when you want to save/share your projects.

**Reply by jayfranks** [reply]
not my fault if thats how you choose to interpret it and the key here is trying to learn not to copy and paste.

**Reply by zjeffer** [reply]
If you want to learn to not copy and paste everything, try writing something simple from scratch. Once you're stuck, google for more information on the exact part you're stuck on, and then try to fix it yourself. Basically: practice!

**Comment by jakobsym** [comment]
This might be of use. 

Resource below, is a great example as why Github showing up in your google searches can be helpful. Maybe the link can be helpful to you in finding direction for your learnings. Good luck with your coding journey! 

https://github.com/ossu/computer-science

**Reply by jayfranks** [reply]
its close to what im looking for but not it. the idea shows your way of thinking is similar but i also need things to be clear and fast paced more project based because that is going to bore the hell out of me hahahaha. i want to get going in kotlin but even there site is flawed and isnt reliable. i dont know anything but i learn fast but i need all the info not what people are deeming necessary. for example i did a test on knowledge i skipped the questions due to lack of knowledge and it then says ive got a good understanding on certain things.......
and im like no ive not pal ...... skipped becuase i dont have clue hahahahaha

**Reply by jayfranks** [reply]
plus the layout is a mess and cant even figure out how navigating all that junk works

**Reply by ABSanthosh** [reply]
You’re saying you are willing to learn but I don’t see it in your answers. If you are asking for specific resources that will help YOU learn, then programming isn’t for you. 
Even tho there are a lot of resources for beginners, you still seem to be struggling to get started. 

In the programming world, you can’t demand others to give you what you need. You have to adopt to what you can find.

you wanted to learn kotlin right? I learned kotlin by building a English to morse code converter from scratch. There isn’t any tutorials for that. Just go figure
